### PR TITLE
UI: Upgrade Icon component

### PR DESCRIPTION
**Pull request by cdedreuille**
## What I did

We introduced a new library for icons. In this PR I'm replacing all icons by the new library. This PR doesn't include the following:
- Icons that have dynamic names
- Icons with styling overrides
- Icons in the legacy component library. Not necessary.

This PR also fixes:
- Removed dependencies in the Storybook icons
- Update the versions of React to make sure it works with the versions of React supported by Storybook.

## How to test

1. Run a sandbox for template, e.g. `yarn task --task sandbox --start-from auto --template react-vite/default-ts`
2. Open Storybook in your browser
3. Access any story and check across addons and the manager if icons are not rendered properly.

## Checklist

- [x] Make sure your changes are tested (stories and/or unit, integration, or end-to-end tests)
- [x] Make sure to add/update documentation regarding your changes
- [x] If you are deprecating/removing a feature, make sure to update
      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

#### Maintainers

- [x] When this PR is ready for testing, make sure to add `ci:normal`, `ci:merged` or `ci:daily` GH label to it to run a specific set of sandboxes. The particular set of sandboxes can be found in `code/lib/cli/src/sandbox-templates.ts`
- [x] Make sure this PR contains **one** of the labels below.

`["cleanup", "BREAKING CHANGE", "feature request", "bug", "build", "documentation", "maintenance", "dependencies", "other"]`

### 🦋 Canary release

This PR does not have a canary release associated. You can request a canary release of this pull request by mentioning the `@storybookjs/core` team here.

_core team members can create a canary release [here](https://github.com/storybookjs/storybook/actions/workflows/canary-release-pr.yml) or locally with `gh workflow run --repo storybookjs/storybook canary-release-pr.yml --field pr=<PR_NUMBER>`_

### CODE REVIEW

**Review by jonniebigodes — APPROVED** [review]
@cdedreuille thanks for taking on this task and updating the snippets we're using. Appreciate it 🙏 ! The documentation changes look good to me, but I'll defer to the rest of the team to see what they say.

So that everyone is aware, I've already followed up with Charles to not only add some changes to the documentation but also a reminder for the addon kit to be updated to factor in these ones as well to prevent the code & documentation fall out of sync.

File: docs/addons/writing-addons.md
Comment (yannbf) [inline docs/addons/writing-addons.md:113]: do we want addon maintainers to do this, given that it's experimental?

File: code/ui/components/src/new/Icon/Icon.mdx
Comment (yannbf) [inline code/ui/components/src/new/Icon/Icon.mdx:17]:
```suggestion
- **Non-decorative** — it means that it delivers information. For example, an icon as only child in a button. The meaning can be obvious visually, but it must have a proper text alternative via `aria-label` for screen readers. (ex: `<Icon.Print aria-label="Print this document" />`)
```

File: code/addons/a11y/src/components/VisionSimulator.tsx
Comment (yannbf) [inline code/addons/a11y/src/components/VisionSimulator.tsx:148]: According to the docs you wrote:
> The meaning can be obvious visually, but it must have a proper text alternative via `aria-label` for screen readers

Should you make sure every icon in this PR has the proper aria-label applied to it?

Comment (cdedreuille) [reply code/addons/a11y/src/components/VisionSimulator.tsx:148]: You're right I could spend some time to do that. That could be a good accessibility win.

Comment (cdedreuille) [reply code/addons/a11y/src/components/VisionSimulator.tsx:148]: @yannbf That would be great but that would require quite some time to test everything. I would put that effort in a separate PR though with other efforts to improve accessibility.

File: docs/addons/writing-addons.md
Comment (cdedreuille) [reply docs/addons/writing-addons.md:113]: I found a new way to help accessibility in a semi automated way. I'll explore this more.

File: code/addons/a11y/src/components/VisionSimulator.tsx
Comment (cdedreuille) [reply code/addons/a11y/src/components/VisionSimulator.tsx:148]: I found a new way to help accessibility in a semi automated way @yannbf. I'll explore this more in a separate PR.

**Review by shilman — APPROVED** [review]
LGTM!

**Pull request merged**
